bindings/pyroot_experimental/PyROOT/python/ROOT/pythonization/_rtensor.py

```diff
@@ -0,0 +1,110 @@
+# Author: Stefan Wunsch CERN  07/2019
+
+################################################################################
+# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.                      #
+# All rights reserved.                                                         #
+#                                                                              #
+# For the licensing terms see $ROOTSYS/LICENSE.                                #
+# For the list of contributors see $ROOTSYS/README/CREDITS.                    #
+################################################################################
+
+from ROOT import pythonization
+from libROOTPython import GetEndianess, GetTensorDataPointer, GetSizeOfType, AsRTensor
+from ROOT.pythonization._rvec import _array_interface_dtype_map
+import cppyy
+
+
+def get_array_interface(self):
+    cppname = type(self).__cppname__
+    for dtype in _array_interface_dtype_map:
+        if not cppname.find("RTensor<{},".format(dtype)) is -1:
+            dtype_numpy = _array_interface_dtype_map[dtype]
+            dtype_size = GetSizeOfType(dtype)
+            endianess = GetEndianess()
+            shape = self.GetShape()
+            strides = self.GetStrides()
+            # Numpy breaks for data pointer of 0 even though the array is empty.
+            # We set the pointer to 1 but the value itself is arbitrary and never accessed.
+            pointer = GetTensorDataPointer(self, cppname)
+            if pointer == 0:
+                pointer == 1
+            return {
+                "shape": tuple(s for s in shape),
+                "strides": tuple(s * dtype_size for s in strides),
+                "typestr": "{}{}{}".format(endianess, dtype_numpy, dtype_size),
+                "version": 3,
+                "data": (pointer, False)
+            }
+
+
+def add_array_interface_property(klass, name):
+    if True in [
+            not name.find("RTensor<{},".format(dtype)) is -1 for dtype in _array_interface_dtype_map
+    ]:
```

Comment:
It is a pity the amount of code repetition we have in `add_array_interface_property` and `get_array_interface` with respect to what is defined in `_rvec.py`. Would it be possible to create two parameterized generic functions and use them both for `RVec` and `RTensor`?

Reply:
Hm I thought about deduplicating here, but the dicts are indeed filled differently (different way to get the pointer, the tensor has strides, ...). So either we reimplement or we have a rather complex if/else construct. I think that the version here is less confusing on the long term.

Reply:
Mm I think this case is on the edge... What about a common helper function that partially fills the dictionary (typestr, version, data), which receives as parameter the function to get the pointer? Then RTensor and RVec would fill what is different in the dictionary (shape and strides).

Reply:
What about this? :smile:

Reply:
Also the data is different :P I'm not really in favor of this.

Reply:
Ok let's make a deal, we leave this one as it is but we refactor more in `PyROOT::AsRTensor` :smile:

Reply:
Deal!